tests/Controllers/DiverControllerTests.cs

```diff
@@ -0,0 +1,368 @@
+            Assert.Equal(model.PersonalBookIssueDate, result.PersonalBookIssueDate);
+            Assert.Equal(model.PersonalBookNumber, result.PersonalBookNumber);
+            Assert.Equal(model.PersonalBookProtocolNumber, result.PersonalBookProtocolNumber);
+            Assert.Equal(model.PhotoUrl, result.PhotoUrl);
+            Assert.Equal(model.Qualification, result.Qualification);
+            Assert.Equal(model.CreatedAt, result.CreatedAt);
+            Assert.Equal(model.UpdatedAt, result.UpdatedAt);
+        }
+
+        [Fact]
+        public async Task GetListAsync_GivenValidInput_ShouldReturnListDiver()
+        {
+            // Arrange
+            var model1 = new Diver()
+            {
+                Address = "г.Ветка, ул.Батракова 32",
+                BirthDate = DateTime.Now,
+                DiverId = 1,
+                FirstName = "Иван",
+                LastName = "Иванов",
+                MedicalExaminationDate = DateTime.Now,
+                MiddleName = "Иванов",
+                PersonalBookIssueDate = DateTime.Now,
+                PersonalBookNumber = "132412",
+                PersonalBookProtocolNumber = "13233434",
+                PhotoUrl = "",
+                Qualification = 1,
+                CreatedAt = DateTime.Now,
+                UpdatedAt = DateTime.Now
+            };
+            var model2 = new Diver()
+            {
+                Address = "г.Ветка, ул.Батракова 32",
+                BirthDate = DateTime.Now,
+                DiverId = 2,
+                FirstName = "Пётр",
+                LastName = "Петров",
+                MedicalExaminationDate = DateTime.Now,
+                MiddleName = "Петрович",
+                PersonalBookIssueDate = DateTime.Now,
+                PersonalBookNumber = "11232",
+                PersonalBookProtocolNumber = "123123",
+                PhotoUrl = "",
+                Qualification = 2,
+                CreatedAt = DateTime.Now,
+                UpdatedAt = DateTime.Now
+            };
+            var modelsFilter = new List<Diver>()
+            {
+                model1
+            };
+            var options = new FilterOptions()
+            {
+                MinHours = 0,
+                MaxHours = 2,
+                MaxQualification = 3,
+                MinQualification = 1,
+                RescueStationId = 127,
+                NameQuery = "Иван"
+            };
+
+            var existingCountOfListItems = 1;
+
+            var diverServiceMock = new Mock<IDiverService>();
+            diverServiceMock.Setup(repo => repo.GetAsync(It.IsAny<FilterOptions>()))
+                .Returns(Task.FromResult((IEnumerable<Diver>)modelsFilter));
+            var diverController = new DiverController(diverServiceMock.Object);
+
+            // Act
+            var result = ((await diverController.GetListAsync(options)) as OkObjectResult).Value as List<Diver>;
```

Comment:
You will have an exception here if the result is not of OkObjectResult - do not combine the checks.

Reply:
This method will return something anyway, even if filter is null.
So I should add a check if it returns null ?